src/transport/SecureSessionMgr.h

```diff
@@ -97,6 +102,17 @@ class EncryptedPacketBufferHandle final : public System::PacketBufferHandle
         return EncryptedPacketBufferHandle(std::move(aBuffer));
     }
 
+    /**
+     * EncryptedPacketBufferHandle represents a buffer we want to resend as-is,
+     * without ever mutating it.  But the actual resend path does need to mutate
+     * it, and more importantly needs to end up calling into sending code that
+     * expects a PacketBuffer.  So we allow converting an
+     * EncryptedPacketBufferHandle to a PacketBufferHandle, in an explicit way.
+     * We only allow doing this with an rvalue reference, so the fact that we
+     * are moving out of the EncryptedPacketBufferHandle is clear.
+     */
+    PacketBufferHandle PrepareToSend() && { return PacketBufferHandle(std::move(*this)); }
```

Comment:
nit: Name of "PrepareToSend" seems to imply usage. Having this as an example of 'we use this when sending' seems good, however I wonder if we could use a different name that is both explicit in what it does but does not explain specific usage patterns.

Thinking maybe `AsSendablePacketBuffer` or `InternalConvertToPacketBuffer`. I have a hard time coming up with a good name as well, just complaining that there is no 'prepare' step in here (just basically a cast/extraction) and sending implies usage (I could take the buffer and store on disk or do a memcompare for something, which are not really sends).

Reply:
> nit: Name of "PrepareToSend" seems to imply usage

Yes.  That's really the only valid use for this function...

I could call this `AsSendablePacketBuffer`, I guess, but I really want it to be clear what this is meant for.

> I could take the buffer and store on disk or do a memcompare for something,

If we end up needing these, we should think about the right sorts of APIs for that.  Just casting to a _writable_  (this part is key!) representation is not the right way to accomplish those goals, right?